base/server/healthcheck/pki/server/healthcheck/meta/csconfig.py

```diff
@@ -55,22 +56,26 @@ def compare_nssdb_with_cs(class_instance, subsystem, cert_tag):
                           nssdbDir=subsystem.instance.nssdb_dir,
                           msg='Unable to load cert from NSSDB: %s' % str(e))
 
-    # Compare whether the certs match
-    if cert_nssdb != cert_cs:
-        directive = '%s.%s.cert' % (subsystem.name, cert_tag)
-        return Result(class_instance, constants.ERROR,
-                      key=cert_id,
-                      nickname=cert['nickname'],
-                      directive=directive,
-                      configfile=subsystem.cs_conf,
-                      msg='Certificate \'%s\' does not match the value '
-                          'of %s in %s' % (cert['nickname'],
-                                           directive,
-                                           subsystem.cs_conf))
-    else:
-        return Result(class_instance, constants.SUCCESS,
-                      key=cert_id,
-                      configfile=subsystem.cs_conf)
+    # base64 blob may contain multiple certs so, compare each in turn
+    cert_split = re.split(b'(Certificate:\n)', cert_nssdb)
```

Comment:
I'll explain what it happening here. If there are 2 certs with the same nickname, this regex will split `cert_nssdb` into 5 parts:

0. Empty string, so we can ignore this
1. "Certificate:\n"
2. <The rest of the first cert>
3. "Certificate:\n"
4. <The rest of the second cert>

So it is necessary to recombine the capture group with the rest of the cert. It is a bit dirty but I couldn't think of a more robust way of doing it - if you can I would be pleased to hear it!

Reply:
I took a different approach, it was easier to do in PEM format. It should be clearer as a result.